launcher/SysInfo.cpp

```diff
@@ -0,0 +1,58 @@
+#include <QString>
+#ifdef Q_OS_MACOS
+#include <sys/sysctl.h>
+#endif
+
+#ifdef Q_OS_MACOS
+bool rosettaDetect() {
+    int ret = 0;
+    size_t size = sizeof(ret);
+    if (sysctlbyname("sysctl.proc_translated", &ret, &size, NULL, 0) == -1)
+    {
+        return false;
+    }
+    if(ret == 0)
+    {
+        return false;
+    }
+    if(ret == 1)
+    {
+        return true;
+    }
+    return false;
+}
+#endif
+
+namespace SysInfo {
+    QString currentSystem() {
+#if defined(Q_OS_LINUX)
+        return "linux";
+#elif defined(Q_OS_MACOS)
+        return "osx";
+#elif defined(Q_OS_WINDOWS)
+        return "windows";
+#elif defined(Q_OS_FREEBSD)
+        return "freebsd";
+#elif defined(Q_OS_OPENBSD)
+        return "openbsd";
+#else
+        return "unknown";
+#endif
+    }
+
+    QString currentArch() {
+        auto qtArch = QSysInfo::currentCpuArchitecture();
+#if defined(Q_OS_MACOS) && !defined(Q_PROCESSOR_ARM)
+        if(rosettaDetect())
+        {
+          return "arm64";
+        }
+        else
+        {
+          return "x86_64";
+        }
+#endif
+        return qtArch;
+    }
```

Comment:
After thinking about this a bit: It doesn't really matter on which architecture PolyMC is running on, but actually on what architecture the JRE is running on. Shouldn't CPU arch be grabbed from there? I.e. someone running a x86-64 JRE on ARM64 should still get x86-64 natives.

Reply:
Done (finally)